Makefile

```diff
@@ -20,7 +20,9 @@ bin: $(shell find . -not -path './vendor/*' -type f -name '*.go') go.mod go.sum
 
 	mkdir -p ./dist
 	rm -rf ./dist/coder-slim_*
+	rm -f ./site/out/bin/coder*
 	./scripts/build_go_slim.sh \
+		--compress 6 \
```

Comment:
What does this line do/why do we need it?

Reply:
From [build_go_slim.sh](https://github.com/coder/coder/blob/c6b1daabc5a7aa67bfbb6c89966d728919ba7f80/scripts/build_go_slim.sh#L19): it compresses the coder-slim binaries.

Reply:
I can try to answer this @jsjoeio.

This is the PR that added the feature #2533, the motivation is to significantly reduce the size of "fat" binaries. We use level 6 for dev builds since it's very fast, 22 for prod builds which shaves an additional 20+MB. (You can see the effect on the releases page, comparing v0.7.3 binaries vs v0.7.4 😄)

I guess it's not strictly necessary to compress for `make bin`, though. Personally I think it's nice if they're prepared in case one does `go build -tags embed` during development, and it's closer to what we'd do with production builds too.

Reply:
Ah! I agree, getting development and production builds to be as similar as possible seems like a good thing to aim for. Thank you both for explaining! Appreciate it 🙌🏼